server/src/main/java/com/sauron/model/BankRequest.java

```diff
@@ -0,0 +1,36 @@
+package com.sauron.model;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+
+public class BankRequest<R> {
+
+    private final HttpMethod httpMethod;
+    private final String path;
+    private final HttpEntity<?> entity;
```

Comment:
Shouldn't be HttpEntity<R>?

Reply:
For the time being I will leave it as it is (to aviod extra casting) but I agree it can't be a wildcard in future.